### DISCUSSION TITLE
BelongsToMany with composite primary key?

### DISCUSSION THREAD

**Discussion by KYSXD** [CodeDredd/pinia-orm#1197] [Q&A] [answered]
Hi!

(Note sure it this has been brought up before)

The current docs states that [Composite Key support](https://pinia-orm.codedredd.de/guide/relationships/getting-started#composite-key-support) works for hasOne, belongsTo & hasMany relations. I'm wondering if belongsToMany also supports this (if so, is something wrong with my code?). If not, is there a way to handle many to many relations with a composite key?.

My models look like these:

```typescript
class Child extends Model {
  static entity = "children";

  static primaryKey = ["field1", "field2"];

  @Str() declare field1: string;
  @Str() declare field2: string;

  @Str() declare foo: string;
};

class Parent extends Model {
  static entity = "parents";

  static primaryKey = "username";

  @Str() declare username: string;

  @BelongsToMany(
    () => Child,
    () => Relation,
    "parentId",
    ["childKey1", "childKey2"],
    "username",
    ["field1", "field2"]
  ) declare related: Child[];
};

class Relation extends Model {
  static entity = "relations";

  static primaryKey = ["parentId", "childKey1", "childKey2"];

  @Str() declare parentId: string;

  @Str() declare childKey1: string;

  @Str() declare childKey2: string;
};
```

So far, after saving a Parent with related (via `save({username: "someRandomIdProvidedByTheUser", related: someArray)`), the parent and all the related are saved. The problem is pivot is not saved. Also this error appears:

```
Error: [Pinia ORM] The record is missing the primary key. If you want to persist record without the primary key, please define the primary key field with the `uid` attribute.
```

And the pivot object is automatically generated as:

```typescript
Object {
  "childKey1,childKey2": undefined,
  parentId: "someRandomIdProvidedByTheUser",
}
```

**Comment by CodeDredd** [comment, marked as answer]
Hey @KYSXD ,

> The current docs states that [Composite Key support](https://pinia-orm.codedredd.de/guide/relationships/getting-started#composite-key-support) works for hasOne, belongsTo & hasMany relations.

That is the state. `BelongsToMany` doesn't support composite key. I thought it wouldn't be used for that and it's compicated to implement so that's why it isn't implemented for `belongToMany` yet. 

A cup of ☕ helps me a lot to be motivated implementing it 😉 if you need it.
